src/main/java/org/ow2/proactive/catalog/service/BucketService.java

```diff
@@ -168,6 +173,16 @@ private List<BucketMetadata> getBucketEntities(List<String> owners, Optional<Str
         return entities;
     }
 
+    private List<String> convertKindFilterToList(Optional<String> kindFilter) {
+        List<String> kindList = new ArrayList<>();
+        if (kindFilter.isPresent()) {
+            kindList = Arrays.asList(kindFilter.get().toLowerCase().split(","));
+        } else {
+            kindList.add("");
```

Comment:
I have just copied this part from the previous code, since it's been used multiple times, I've just moved it into a separate function. 
I agree it's a bit strange. My guess is that the previous code prefers to have a list with an empty kind filter than an empty list to avoid a case check in the part you referred to.